### PR TITLE
Change setup to pyproject

### PR DESCRIPTION
**Pull request by marco-2023**
@RichRick1: Added the requested features in #55 

As it is now:

```bash
pip install -e .
```
will install package and basic dependencies. Then the extra dependencies can be installed by
```bash
pip install -e .[gui]
```
will install `tkinter`
```bash
pip install -e .[gpt]
```
or
```bash
pip install -e .[toml]
```
will install toml

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+Issues = "https://github.com/theochem/ModelHamiltonian/issues"
+Source = "https://github.com/theochem/ModelHamiltonian/"
+
+[project.optional-dependencies]
+gui = [
+    "tkinter",
+]
+
+gpt = [
+    "toml"
```
Comment (RichRick1) [inline pyproject.toml:47]: can we add openai here too?

**Review by RichRick1 — CHANGES_REQUESTED** [review]
It looks great to me, thanks!
Can we add openai dependency here? For the gpt installation

File: pyproject.toml (hunk 1 above)
Comment (marco-2023) [reply pyproject.toml]: Done! I did not see that one. My bad.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+]
+dynamic =["version" ]
+
+[project.urls]
+Documentation = "https://moha.qcdevs.org"
+Issues = "https://github.com/theochem/ModelHamiltonian/issues"
+Source = "https://github.com/theochem/ModelHamiltonian/"
+
+[project.optional-dependencies]
+gui = [
```
Comment (RichRick1) [inline pyproject.toml:42]: we also need to support rdkit in here too
Sorry, this part is growing fast

**Review by RichRick1 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (marco-2023) [reply pyproject.toml:42]: Hi @RichRick1 I added `rdkit` to the GUI dependencies. Please note that I removed `tkinter`, this is part of the python standard library so there is no need to install it.

**Review by RichRick1 — APPROVED** [review]

**Pull request merged**